system/lib/compiler-rt/lib/sanitizer_common/sanitizer_platform_interceptors.h

```diff
@@ -308,7 +308,7 @@ SANITIZER_WEAK_IMPORT void *aligned_alloc(__sanitizer::usize __alignment,
 #define SANITIZER_INTERCEPT_SENDMMSG SI_LINUX
 #define SANITIZER_INTERCEPT_SYSMSG SI_LINUX_NOT_ANDROID
 #define SANITIZER_INTERCEPT_GETPEERNAME SI_POSIX
-#define SANITIZER_INTERCEPT_IOCTL SI_POSIX && !SI_EMSCRIPTEN
+#define SANITIZER_INTERCEPT_IOCTL SI_POSIX_NOT_EMSCRIPTEN
```

Comment:
Why do we have a special define for this but not, for example, for  `SI_POSIX && !SI_SOLARIS` above?
`

Reply:
Good question, I'm not  sure. I attempted to unify these checks somewhat, since we use `SI_POSIX_NOT_EMSCRIPTEN` elsewhere in this file, but I'm happy to split it out if you'd prefer.

Reply:
Oh. I see, its pre-existing.  LGTM then, although I'm not sure why we added this since its only used in one place right now.